hepdata/modules/records/utils/analyses.py

```diff
@@ -50,207 +50,219 @@ def test_analyses_schema(json_file, schema_version="1.0.0"):
+                # Check for missing analyses.
+                for ana in r_json["analyses"]:
+                    inspire_id = str(ana["inspire_id"])  # inspire_id is stored as a string in the database
+                    submission = get_latest_hepsubmission(inspire_id=inspire_id, overall_status='finished')
 
-                                try:
-                                    db.session.add(submission)
-                                    db.session.commit()
-                                    latest_submission = get_latest_hepsubmission(inspire_id=record)
-                                    if submission.version == latest_submission.version:
-                                        index_record_ids([submission.publication_recid])
-                                except Exception as e:
-                                    db.session.rollback()
-                                    log.error(e)
+                    if submission:
+                        num_new_resources = 0
 
-                        else:
-                            log.debug("An analysis is available in {0} but with no equivalent in HEPData (ins{1}).".format(
-                                analysis_endpoint, record))
+                        for implementation in ana["implementations"]:
+                            _resource_url = r_json["url_templates"]["main_url"].format(**implementation)
 
-                else:  # schema_version >= "1.0.0"
+                            if not is_resource_added_to_submission(submission.publication_recid, submission.version,
+                                                                _resource_url):
+
+                                log.info('Adding {} analysis to ins{} with URL {}'.format(
+                                    analysis_endpoint, inspire_id, _resource_url)
+                                )
+                                new_resource = DataResource(
+                                    file_location=_resource_url,
+                                    file_type=analysis_endpoint,
+                                    file_description=r_json["implementations_description"]
+                                )
+
+                                if "implementations_license" in r_json:
+                                    resource_license = get_license(r_json["implementations_license"])
+                                    new_resource.file_license = resource_license.id
+
+                                submission.resources.append(new_resource)
+                                num_new_resources += 1
+
+                            else:
+
+                                # Remove resources from 'analysis_resources' list.
+                                resources = list(filter(lambda a: a.file_location == _resource_url, analysis_resources))
+                                for resource in resources:
+                                    analysis_resources.remove(resource)
```

Comment:
These lines don't seem to be covered by tests:
https://coveralls.io/jobs/172976690/source_files/8867976225#L174